src/Components/WebAssembly/Server/src/ComponentsWebAssemblyApplicationBuilderExtensions.cs

```diff
@@ -47,6 +47,15 @@ public static IApplicationBuilder UseBlazorFrameworkFiles(this IApplicationBuild
                 {
                     context.Response.Headers.Append("Blazor-Environment", webHostEnvironment.EnvironmentName);
 
+                    // DOTNET_MODIFIABLE_ASSEMBLIES is used by the runtime to initialize hot-reload specific environment variables and is configured
+                    // by the launching process (dotnet-watch / Visual Studio).
+                    // In Development, we'll transmit the environment variable to WebAssembly as a HTTP header. The bootstrapping code will read the header
+                    // and configure it as env variable for the wasm app.
+                    if (webHostEnvironment.IsDevelopment() && Environment.GetEnvironmentVariable("DOTNET_MODIFIABLE_ASSEMBLIES") is { } modifiableAssemblies)
```

Comment:
What does `Environment.GetEnvironmentVariable("DOTNET_MODIFIABLE_ASSEMBLIES") is { }` mean

Reply:
It means that the expression is not null, without constraints on the properties. From https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/proposals/csharp-8.0/patterns#property-pattern, 

```
if (s is {} x) ... // x is of type string
```

Reply:
It is incredibly obscure, I understood what it was doing. Can we please replace it with something "less fancy". Features like these raise the bar for contributors that need to learn about them when there are equally valid alternatives.

Reply:
I wanted to avoid reading the Env variable in non-development environments. We might eventually get a runtime flag for determining HotReload, but we could use that here.